File: app/src/main/java/sherafatpour/entekhab/notify/NotificationRepository.java
package sherafatpour.entekhab.notify;

import android.app.Application;
import android.os.AsyncTask;

import androidx.lifecycle.LiveData;

import java.util.List;

public class NotificationRepository {
    private NotificationDao notificationDao;
    private LiveData<List<Notification>> allNote;

    public NotificationRepository(Application application) {
        NotificationDatabase database = NotificationDatabase.getInstance(application);
        notificationDao = database.noteDao();
        allNote = notificationDao.getAllNote();
    }

    public void insert(Notification note) {
        new InsertNoteAsyncTask(notificationDao).execute(note);
    }

    public void update(Notification note) {
    new UpdateNoteAsyncTask(notificationDao).execute(note);
    }

    public void delete(Notification note) {
        new DeleteNoteAsyncTask(notificationDao).execute(note);

    }

    public void deleteAllNotes() {
        new DeleteAllNoteAsyncTask(notificationDao).execute();

    }

    public LiveData<List<Notification>> getAllNotes() {
        return allNote;
    }


    private static class InsertNoteAsyncTask extends AsyncTask<Notification, Void, Void> {
        private NotificationDao notificationDao;

        private InsertNoteAsyncTask(NotificationDao notificationDao) {
            this.notificationDao = notificationDao;
        }

        @Override
        protected Void doInBackground(Notification... notes) {
            notificationDao.insert(notes[0]);
            return null;
        }
    }

    private static class UpdateNoteAsyncTask extends AsyncTask<Notification, Void, Void> {
        private NotificationDao notificationDao;

        private UpdateNoteAsyncTask(NotificationDao notificationDao) {
            this.notificationDao = notificationDao;
        }

        @Override
        protected Void doInBackground(Notification... notes) {
            notificationDao.update(notes[0]);
            return null;
        }
    }

    private static class DeleteNoteAsyncTask extends AsyncTask<Notification, Void, Void> {
        private NotificationDao notificationDao;

        private DeleteNoteAsyncTask(NotificationDao notificationDao) {
            this.notificationDao = notificationDao;
        }

        @Override
        protected Void doInBackground(Notification... notes) {
            notificationDao.delete(notes[0]);
            return null;
        }
    }


    private static class DeleteAllNoteAsyncTask extends AsyncTask<Notification, Void, Void> {
        private NotificationDao notificationDao;

        private DeleteAllNoteAsyncTask(NotificationDao notificationDao) {
            this.notificationDao = notificationDao;
        }

        @Override
        protected Void doInBackground(Notification... notes) {
            notificationDao.deleteAllNote();
            return null;
        }
    }

}

File: app/src/main/java/sherafatpour/entekhab/notify/room/NotificationRepository.kt
package sherafatpour.entekhab.notify.room

import android.app.Application
import android.os.AsyncTask

import androidx.lifecycle.LiveData

import sherafatpour.entekhab.notify.model.Notification

class NotificationRepository(application: Application) {
    private val notificationDao: NotificationDao
    val allUnReadNotification: LiveData<List<Notification>>
    val allReadNotification: LiveData<List<Notification>>

    init {
        val database = NotificationDatabase.getInstance(application)
        notificationDao = database.noteDao()
        allUnReadNotification = notificationDao.allUnreadNotification
        allReadNotification = notificationDao.allReadNotification
    }

    fun insert(note: Notification) {
        InsertNoteAsyncTask(notificationDao).execute(note)
    }

    fun update(note: Notification) {
        UpdateNoteAsyncTask(notificationDao).execute(note)
    }

    fun delete(note: Notification) {
        DeleteNoteAsyncTask(notificationDao).execute(note)

    }

    fun deleteAllNotes() {
        DeleteAllNoteAsyncTask(notificationDao).execute()
    }

    fun updateStatus(id:Int) {
        UpdateStatusAsyncTask(notificationDao).execute(id)
    }


    private class InsertNoteAsyncTask(private val notificationDao: NotificationDao) : AsyncTask<Notification, Void, Void>() {

        override fun doInBackground(vararg notes: Notification): Void? {
            notificationDao.insert(notes[0])
            return null
        }
    }

    private class UpdateNoteAsyncTask(private val notificationDao: NotificationDao) : AsyncTask<Notification, Void, Void>() {

        override fun doInBackground(vararg notes: Notification): Void? {
            notificationDao.update(notes[0])
            return null
        }
    }

    private class DeleteNoteAsyncTask(private val notificationDao: NotificationDao) : AsyncTask<Notification, Void, Void>() {

        override fun doInBackground(vararg notes: Notification): Void? {
            notificationDao.delete(notes[0])
            return null
        }
    }

    private class DeleteAllNoteAsyncTask(private val notificationDao: NotificationDao) : AsyncTask<Notification, Void, Void>() {

        override fun doInBackground(vararg notes: Notification): Void? {
            notificationDao.deleteAllNote()
            return null
        }
    }


    private class UpdateStatusAsyncTask(private val notificationDao: NotificationDao) : AsyncTask<Int, Void, Void>() {
        override fun doInBackground(vararg params: Int?): Void? {
            notificationDao.updateStatus(params[0]!!)
            return null

        }


    }

}
